src/java.base/share/classes/sun/security/provider/MD4.java

```diff
@@ -78,13 +78,8 @@ public Void run() {
         });
     }
 
-    public static MessageDigest getInstance() {
```

Comment:
Just let it throw the NSAE. There are 2 usages of this method, one will throw an `AssertionError` and one below in `ArcFourCrypto` will propagate it and finally it will get caught in `EncrptionKey::stringToKey` like all other crypto-related exceptions.

Reply:
nit: copyright year change to 2024?

Reply:
Missed copyright year update of this file? The other 2 are updated.

Reply:
Actually, some other commits have already updated it. I merge from master and now it shows 2024.